src/twisted/web/server.py

```diff
@@ -878,3 +884,110 @@ def acceptableProtocols(self):
             baseProtocols.insert(0, b"h2")
 
         return baseProtocols
+
+
+class _HTTPSRedirectProtocolWrapper(ProtocolWrapper):
+    """
+    Wraps the TLS protocol and redirect to HTTPS if an HTTP request is
+    detected after a connection is made.
```

Comment:
I feel like this doesn't really cover the inherently potentially-broken nature of the one-reassembled-TCP-segment heuristic being used here.  (Or maybe that's more of an issue for the public docstring, but it's not covered there, either...)